### PR TITLE
[RFC] Include 'j' in default '&formatoptions'

### PR DESCRIPTION
**Pull request by fmoralesc**
Re: https://github.com/neovim/neovim/issues/1664

### CODE REVIEW

**Comment by Pyrohh** [comment]
I noticed that under `fo-table`, `tcq` still says it's the default. Perhaps that should just be removed, as it can most likely get out of date easily.

**Comment by fmoralesc** [comment]
@Pyrohh thanks for pointing it out. I'll remove that sentence in this PR, I agree that's for the best.

File: runtime/doc/vim_diff.txt [hunk 1]
```diff
@@ -30,6 +30,7 @@ these differences.
 
 - 'backspace' defaults to "indent,eol,start"
 - 'encoding' defaults to "utf-8"
+- 'formatoptions'  includes "j" (see |fo-table|)
```
Comment (Pyrohh) [inline runtime/doc/vim_diff.txt]: Maybe just `formatoptions default to "tcqj"`, for consistency with `backspace` above.
I don't know about you, but I prefer the absolute definition over the relative one (which requires people to type `:help formatoptions`).

Comment (fmoralesc) [inline runtime/doc/vim_diff.txt]: Sure, [I don't have strong feelings one way or the other](https://www.youtube.com/watch?v=CxK_nA2iVXw)

Comment (Pyrohh) [inline runtime/doc/vim_diff.txt]: Tel my wife I said "hello".

**Comment by Pyrohh** [comment]
Besides a rebase, LGTM.

File: runtime/doc/options.txt [hunk 2]
```diff
@@ -2912,7 +2912,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	evaluating 'foldtext' |textlock|.
 
 					*'formatoptions'* *'fo'*
-'formatoptions' 'fo'	string (Vim default: "tcq", Vi default: "vt")
+'formatoptions' 'fo'	string (Vim default: "tcqj", Vi default: "vt")
```
Comment (fwalch) [inline runtime/doc/options.txt]: Neovim default? Or: `Neovim default: "tcqj", Vim default: "tcq", Vi default: "vt"`?

File: runtime/doc/vim_diff.txt [hunk 3]
```diff
@@ -30,6 +30,7 @@ these differences.
 
 - 'backspace' defaults to "indent,eol,start"
 - 'encoding' defaults to "utf-8"
+- 'formatoptions'  defaults to "tcqj"
```
Comment (fwalch) [inline runtime/doc/vim_diff.txt]: There are two spaces between `'formatoptions'` and `defaults`.

File: runtime/doc/options.txt (hunk 2 above)
Comment (justinmk) [inline runtime/doc/options.txt]: In other PRs we simply used "default". I think that is good enough.

Comment (fmoralesc) [inline runtime/doc/options.txt]: I'll use 'default', we don't have a `set -&nvim` idiom.

File: runtime/doc/vim_diff.txt (hunk 3 above)
Comment (fmoralesc) [inline runtime/doc/vim_diff.txt]: :+1:

**Comment by Pyrohh** [comment]
I think this change is a great idea, but that's just me. @fwalch @justinmk?

**Comment by justinmk** [comment]
LGTM. Merged

**Pull request closed without merging**